src/runtime/cpu_device_api.cc

```diff
@@ -63,6 +65,11 @@ class CPUDeviceAPI final : public DeviceAPI {
         static_cast<const char*>(from) + from_offset, size);
   }
 
+  void RecordedCopyDataFromTo(
+      void* from, size_t from_offset, void* to, size_t to_offset, size_t size,
+      DGLContext ctx_from, DGLContext ctx_to, DGLDataType type_hint,
+      void* pyt_ctx) final {}
```

Comment:
Should we raise an exception if this is called?

Reply:
I agree.  This function should never be called and therefore it's better to call `BUG_IF_FAIL`.

Reply:
Updated.